### PR TITLE
chore: backports from v0.24-dev.30 to v0.24.0

### PR DESCRIPTION
**Pull request by antouhou**
## What was done?

Backports satrting from v0.24-dev.30 to v0.24.0

### Skipped js-drive PRs
- https://github.com/dashpay/platform/pull/1034
- https://github.com/dashpay/platform/pull/1045

## How Has This Been Tested?

- Run tests

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: packages/wasm-dpp/scripts/build-wasm.sh [hunk 1]
```diff
@@ -25,10 +25,10 @@ if ! [[ -d $OUTPUT_DIR ]];  then
   mkdir -p "$OUTPUT_DIR"
 fi
 
-if ! [ -x "$(command -v wasm-bindgen)" ]; then
-    echo 'Wasm-bindgen CLI is not installed. Installing';
-    cargo install -f wasm-bindgen-cli
-fi
+#if ! [ -x "$(command -v wasm-bindgen)" ]; then
+#    echo 'Wasm-bindgen CLI is not installed. Installing';
+    cargo install -f wasm-bindgen-cli@0.2.84
```
Comment (markin-io) [inline packages/wasm-dpp/scripts/build-wasm.sh:30]: Hm. Is it going to be re-installed on every script run? Or will cargo detect that this version is already installed and just wont do anything?

File: packages/wasm-dpp/src/identity/state_transition/transition_types.rs [hunk 2]
```diff
@@ -0,0 +1,30 @@
+use dpp::state_transition::StateTransitionType;
+use wasm_bindgen::prelude::*;
+
+#[wasm_bindgen(js_name=StateTransitionTypes)]
+#[allow(non_camel_case_types)]
+pub enum StateTransitionTypeWasm {
```
Comment (markin-io) [inline packages/wasm-dpp/src/identity/state_transition/transition_types.rs:6]: 👍

File: packages/wasm-dpp/src/protocol_version.rs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+use dpp::version::LATEST_VERSION;
+use wasm_bindgen::prelude::wasm_bindgen;
+
+#[wasm_bindgen(js_name = getLatestProtocolVersion)]
+pub fn latest_protocol_version() -> u32 {
+    LATEST_VERSION
```
Comment (markin-io) [inline packages/wasm-dpp/src/protocol_version.rs:6]: 👍

File: packages/wasm-dpp/scripts/build-wasm.sh (hunk 1 above)
Comment (antouhou) [reply packages/wasm-dpp/scripts/build-wasm.sh:30]: It is going to. The reason is that no matter what I did if I didn't forcibly reinstall it on every run, something broke in the CI. I mentioned it on yesterday's standup

Comment (markin-io) [reply packages/wasm-dpp/scripts/build-wasm.sh:30]: Agreed on fixing this in the next PR

**Review by markin-io — APPROVED** [review]

**Review by shumkov — APPROVED** [review]

**Pull request merged**